index.js

```diff
@@ -9,16 +9,14 @@ const takeUntil = sSrc => src => (start, sink) => {
       sourceTalkback = d;
 
       sSrc(0, (st, sd) => {
-        if (done) return;
-
         if (st === 0) {
           sTalkback = sd;
           sTalkback(1);
           return
         }
         done = true;
-        sourceTalkback(2);
         sTalkback(2);
```

Comment:
this reordering might bot be relevant, but it just made more sense to me to unsubscribe from the notifier source first and then propagate that information to source & sink